src/protocols/secure_channel/PASESession.cpp

```diff
@@ -758,7 +758,7 @@ void PASESession::OnMessageReceived(ExchangeContext * ec, const PacketHeader & p
 
     if (mExchangeCtxt == nullptr)
     {
-        mExchangeCtxt = ec->Retain();
+        mExchangeCtxt = ec;
```

Comment:
This condition only triggers when the message is received via unsolicited message handler. In that scenario, the exchange manager creates a new exchange context and hands over the reference of it to the callee.

Reply:
Again, it would be good to document, on `ExchangeDelegateBase`, the semantics: that if the delegate is used for an UnsolicitedMessageHandler the exchange context will be allocated before the first `OnMessageReceived` call for a given exchange and it's the delegate's job to `Close` it when done, handle multiple exchanges, etc.

It might be cleaner, in the long-term, to have an explicit "new ExchangeContext created" notification....  Maybe.  Need to think about it.

Reply:
Added the documentation.